spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaClientAutoConfiguration.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.spring;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.WebClientBuilder;
+
+/**
+ * An auto-configuration for {@link WebClient}.
+ */
+@Configuration
+@ConditionalOnClass(WebClient.class)
+public class ArmeriaClientAutoConfiguration extends AbstractArmeriaClientAutoConfiguration {
+
+    /**
+     * Create an {@link WebClientBuilder} bean.
+     */
+    @Bean
+    @Scope("prototype")
```

Comment:
Is prototype scope better than returning `Supplier<WebClientBuilder>`? It seems like it could be more limiting, e.g., if you inject `WebClientBuilder` you can only create one client in a given injected method?

Reply:
I don't know if I can say that it's better to prototype the bean scope. 🤔 
However, if it's more restrictive, I'll check out what's suggested. 👍

Reply:
I don't rely on Spring so I think it's okay. 
I have corrected this. Thank you for the suggestion. 👍